### DISCUSSION TITLE
Intl.DateTimeFormat wrong locale

### DISCUSSION THREAD

**Discussion by vasco3** [community/community#26860] [Actions]
<p>I’m using github actions to run jest tests. But the action isnt’ recognizing that I’m setting  Intl.DateTimeFormat to es-MX instead of en-US.</p>
<p>I see that there might be the case that the github actions container has less locales available. Is there a way to use es-MX?</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/vasco3">@vasco3</a> ,</p>
<p>What runner did you use? GitHub-hosted runner or your self-hosted runner?</p>
<p>On GitHub-hosted runner, you can try the following commands to see if the system local can be changed:</p>
<ul>
<li>
<p>On Linux</p>
<pre><code class="lang-auto"># View information of current System Locale
locale

# Set LANG to es-MX.UTF-8
sudo update-locale LANG=LANG=es-MX.UTF-8 LANGUAGE

# Configure the locale parameter LC_TIME
sudo update-locale LC_TIME=es-MX.UTF-8
</code></pre>
</li>
<li>
<p>On Windows</p>
<pre><code class="lang-powershell">Set-WinSystemLocale -SystemLocale es-MX
</code></pre>
</li>
<li>
<p>On macOS, using the “<strong>sudo languagesetup</strong>” command.</p>
</li>
</ul>
<br>
<p>On the self-hosted runner, you can try to set the system local of the runner machine as <strong>Spanish (Mexico)</strong>, also known as <strong>es-MX</strong>.</p>

**Comment by vasco3** [comment]
<p>this is my setup and it is a Github-hosted runner</p>
<pre><code class="lang-auto">name: Unit Tests

on:
  push:
    branches:
      - master
  pull_request:
    branches: [master]

jobs:
  Jest:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [14.8]

    steps:
      - uses: actions/checkout@v1
      - name: Run Tests
        run: |
          npm install 
          npm test

</code></pre>
<p><code>npm test</code> runs well in my local, the problem is when it runs as a Github action.</p>
<p>not sure how could I run the <code>update-locale</code> command in yaml</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/vasco3">@vasco3</a> ,</p>
<p>I have created an issue ticket (<a href="https://github.com/actions/virtual-environments/issues/1650" rel="noopener nofollow ugc">actions/virtual-environments#1650</a>) to help you report the problem to the appropriate engineering team for further investigation and evaluation.<br>
You can follow this issue ticket and add your comments to it.</p>

**Comment by vasco3** [comment]
<p>thank you, I will follow up on there</p>

**Comment by al-cheb** [comment, marked as answer]
<p>Hello, <a class="mention" href="/vasco3">@vasco3</a><br>
I can reproduce the issue with Node 12 which one set up by default. The official distribution of Node 14 includes ‘full icu’: the locale database. Without that data Node only supports ‘en-US’ and will throw for other languages, and silently fall back to ‘en-US’ for English with other countries.</p>
<pre><code class="lang-auto">jobs:
  build:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/setup-node@v2-beta
        with:
          node-version: '14'
      - name: node
        run: |
          cat &lt;&lt;EOF &gt; test.js
          const date = new Date(Date.UTC(2012, 11, 20, 3, 0, 0)); 
          console.log(new Intl.DateTimeFormat('es-MX').format(date));  
          console.log(new Date(date).toLocaleString('es-MX', {weekday:'long'}))
          EOF
          node test.js
</code></pre>
<p><img src="https://user-images.githubusercontent.com/47745270/181100339-be8eb45c-9ba0-446c-845a-252652d4c4d4.png" alt="image" data-base62-sha1="9Vb5yGQjneb98gRqvlx7Qdk7TEY" width="467" height="192"></p>

**Comment by vasco3** [comment]
<aside class="quote no-group" data-username="vasco3" data-post="3" data-topic="133337">
<div class="title">
<div class="quote-controls"></div>
 vasco3:</div>
<blockquote>
<pre><code class="lang-auto">    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [14.8]
</code></pre>
</blockquote>
</aside>
<p>Thank you <a class="mention" href="/al-cheb">@al-cheb</a> this is helpful. I’ll use this instead</p>

**Comment by mentalsolin** [comment]
I updated Node to version 16 and everything worked for me. Thanks
